modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/GoogleSpeechAttachTranscriptionOperationHandler.java

```diff
@@ -0,0 +1,178 @@
+    CONFIG_OPTIONS.put(TARGET_CAPTION_FORMAT, "The target caption format of the transcription file (vtt, dfxp, etc)");
+    CONFIG_OPTIONS.put(TRANSCRIPTION_LINE_SIZE, "Line size of transcription text to display on video");
+  }
+
+  @Override
+  protected void activate(ComponentContext cc) {
+    super.activate(cc);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see
+   * org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance,
+   * JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+
+    logger.debug("Attach transcription for mediapackage {} started", mediaPackage);
+
+    // Get job id.
+    String jobId = StringUtils.trimToNull(operation.getConfiguration(TRANSCRIPTION_JOB_ID));
+    if (jobId == null) {
+      throw new WorkflowOperationException(TRANSCRIPTION_JOB_ID + " missing");
+    }
+
+    // Check which tags/flavors have been configured
+    String targetTagOption = StringUtils.trimToNull(operation.getConfiguration(TARGET_TAG));
+    String targetFlavorOption = StringUtils.trimToNull(operation.getConfiguration(TARGET_FLAVOR));
+    // Target flavor is mandatory
+    if (targetFlavorOption == null) {
+      throw new WorkflowOperationException(TARGET_FLAVOR + " missing");
+    }
+    MediaPackageElementFlavor flavor = MediaPackageElementFlavor.parseFlavor(targetFlavorOption);
+    String captionFormatOption = StringUtils.trimToNull(operation.getConfiguration(TARGET_CAPTION_FORMAT));
+
+    // Get line size if set
+    String lineSize = StringUtils.trimToNull(operation.getConfiguration(TRANSCRIPTION_LINE_SIZE));
+    if (lineSize == null) {
+      lineSize = "100"; // Use default line size
+    }
+
+    try {
+      // Get transcription file from the service
+      MediaPackageElement original = service.getGeneratedTranscription(mediaPackage.getIdentifier().compact(), jobId);
+      MediaPackageElement transcription = original;
+
+      // If caption format passed, convert to desired format
+      if (captionFormatOption != null) {
+        Job job = captionService.convert(transcription, "google-speech", captionFormatOption, lineSize);
+        if (!waitForStatus(job).isSuccess()) {
+          throw new WorkflowOperationException("Transcription format conversion job did not complete successfully");
+        }
+        transcription = MediaPackageElementParser.getFromXml(job.getPayload());
+      }
+
+      // Set the target flavor
+      transcription.setFlavor(flavor);
+
+      // Add tags
+      if (targetTagOption != null) {
+        for (String tag : asList(targetTagOption)) {
+          if (StringUtils.trimToNull(tag) != null) {
```

Comment:
`!StringUtils.isBlank` would be a better fit here.

Reply:
[`StringUtils.isNotBlank(…)`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotBlank-java.lang.CharSequence-) ;-)

Reply:
@lkiesow has a better solution :)

Reply:
fixed